worker/docker_image_manager.py

```diff
@@ -0,0 +1,135 @@
+import threading
+import os
+import time
+import traceback
+import logging
+import json
+
+import file_util
+
+logger = logging.getLogger(__name__)
+
+
+class DockerImageManager(object):
+    """
+    Manages the Docker images available on the worker, and ensures that they
+    do not fill up the disk.
+
+    Warning: Assumes that Docker images are stored on the same disk as the
+    work directory specified by the `workdir` parameter. If this is not the
```

Comment:
Actually I just realized it's simply impossible on Mac, so there's no reason to support it anyway.
Basically, Docker for Mac doesn't free up disk space when you delete images, until you restart the whole thing. It has something to do with the fact that it's running in a VM.

I'll just point at `/var/lib/docker/aufs/diff` (which holds the images) then!